### PR TITLE
Add stubs for RPi.GPIO

### PR DESCRIPTION
**Pull request by fperrin**
Hi,

>  RPi.GPIO allows controlling Raspberry Pi GPIO channels in Python.
>
> It provides all the basic functionality, but is unsuitable for real-time or timing critical applications. RPi.GPIO also does not support SPI, I²C or hardware PWM yet. 

These stubs cover the entirety of the API exposed to Python.

The [name of the project on PyPi](https://pypi.org/project/RPi.GPIO/) is `RPi.GPIO` with a `.`, hopefully that won't cause any issue.

There is [very little activity upstream](https://sourceforge.net/p/raspberry-gpio-python/code/ci/be8e4d5223489838b9dfe05b3a25587f8891a5db/log/?path=), but that module is still useful, indeed it comes by default with Raspbian as provided with brand new Pi modules.

### CODE REVIEW

File: stubs/RPi.GPIO/RPi/GPIO/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from collections.abc import Callable
+from typing import Any, Literal
+from typing_extensions import TypeAlias
+
+VERSION: str
+RPI_INFO: dict[str, Any]
```
Comment (srittau) [inline stubs/RPi.GPIO/RPi/GPIO/__init__.pyi:6]: This could be a `TypedDict`, but this can always be done later.

**Review by srittau — APPROVED** [review]
Thanks, LGTM. Let's see whether it breaks our stub uploader.

File: stubs/RPi.GPIO/RPi/GPIO/__init__.pyi (hunk 1 above)
Comment (fperrin) [reply stubs/RPi.GPIO/RPi/GPIO/__init__.pyi:6]: Thank you for the review @srittau 

I just tried that, and mypy failed with: https://github.com/fperrin/typeshed/actions/runs/7710857960/job/21015067471?pr=3

> mypy --platform win32 --python-version 3.9 on the test cases for 'requests': FAILURE
> ...
>   File "mypy/fixup.py", line 127, in visit_symbol_table
>   File "mypy/lookup.py", line 31, in lookup_fully_qualified
> AssertionError: Cannot find module for resource.struct_rusage

I really doubt this is related to my change, but who knows...

Comment (AlexWaygood) [reply stubs/RPi.GPIO/RPi/GPIO/__init__.pyi:6]: @fperrin sorry about that! That's entirely unrelated, and it's very much on our minds right now: see

- #11220

Comment (fperrin) [reply stubs/RPi.GPIO/RPi/GPIO/__init__.pyi:6]: in that case I'll ignore that failure and push Sebastian's suggestion

**Comment by Avasam** [comment]
Cross-referencing the following for discoverability:
- https://sourceforge.net/p/raspberry-gpio-python/tickets/178/
- https://github.com/microsoft/python-type-stubs/issues/207

**Pull request merged**